cairo/tests/ethereum/cancun/test_fork.py

```diff
@@ -91,11 +127,30 @@ def tx_without_code(draw):
 
 @composite
 def tx_with_sender_in_state(
-    draw, state_strategy=state, account_strategy=account_strategy
+    draw,
+    # We need to set a high tx gas so that tx.gas > intrinsic cost and base_fee < gas_price
+    tx_strategy=tx_with_small_data(
+        gas_strategy=st.integers(min_value=2**16, max_value=2**26 - 1).map(Uint),
+        gas_price_strategy=st.integers(
+            min_value=MIN_BASE_FEE - 1, max_value=2**64
+        ).map(Uint),
+    ),
+    state_strategy=state,
+    account_strategy=account_strategy,
+    probabilities=80,
 ):
     state = draw(state_strategy)
-    chain_id = draw(st.from_type(U64))
-    tx = draw(st.from_type(Transaction))
+    account = draw(account_strategy)
+    # 2 * chain_id + 35 + v must be less than 2^64 for the signature of a legacy transaction to be valid
+    chain_id = draw(st.integers(min_value=1, max_value=(2**64 - 37) // 2).map(U64))
+    tx = draw(tx_strategy)
+    nonce = U256(account.nonce)
+    # To avoid useless failures, set nonce of the transaction to the nonce of the sender account
+    tx = (
+        replace(tx, chain_id=chain_id, nonce=nonce)
+        if not isinstance(tx, LegacyTransaction)
+        else replace(tx, nonce=nonce)
+    )
```

Comment:
get chain_id and nonce from tx

Reply:
Might not be possible not possible ?
Nonce is take from the sender account and chain_id is modified to ensure is it valid for legacytx signatures